src/main/java/org/jabref/logic/ai/embeddings/AiEmbeddingsManager.java

```diff
@@ -0,0 +1,96 @@
+package org.jabref.logic.ai.embeddings;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.desktop.JabRefDesktop;
+import org.jabref.preferences.AiPreferences;
+
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.filter.MetadataFilterBuilder;
+import org.h2.mvstore.MVStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is responsible for managing the embeddings cache. The cache is saved in a local user directory.
+ * <p>
+ * MVStore is used as an embedded database. It stores the embeddings and what files have been fully ingested.
+ * {@link org.jabref.model.entry.LinkedFile} and embeddings are connected with LinkedFile.getLink().
+ * <p>
+ * In case an error occurs while opening an MVStore, the class will notify the user of this error and continue
+ * with in-memory store (meaning all embeddings will be thrown away on exit).
+ * <p>
+ * This class also listens for changes of embeddings parameters (in AI "Expert settings" section). In case any of them
+ * changes, the embeddings should be invalidated (cleared).
+ */
+public class AiEmbeddingsManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AiEmbeddingsManager.class);
+
+    private static final String EMBEDDINGS_STORE_FILE_NAME = "embeddings.mv";
+
+    private final AiPreferences aiPreferences;
+
+    private final MVStore mvStore;
+
+    private final MVStoreEmbeddingStore embeddingStore;
+    private final AiIngestedFilesTracker ingestedFilesTracker;
+
+    public AiEmbeddingsManager(AiPreferences aiPreferences, DialogService dialogService) {
+        this.aiPreferences = aiPreferences;
+
+        Path mvStorePath = JabRefDesktop.getEmbeddingsCacheDirectory().resolve(EMBEDDINGS_STORE_FILE_NAME);
+
+        try {
+            Files.createDirectories(JabRefDesktop.getEmbeddingsCacheDirectory());
+        } catch (IOException e) {
+            dialogService.showErrorDialogAndWait("An error occurred while creating directories for embedding store. Will use an in-memory store", e);
+            mvStorePath = null;
+        }
+
+        MVStore mvStoreTemp; // Strange Java...
```

Comment:
Java tries to have proper variable scoping rules. Be happy that you did not grew up in times where variables could be declared after use.

```suggestion
        MVStore mvStoreTemp;
```

Reply:
After a little bit of thought I understand now why Java forbids this (even if in this concrete example it's fine)